tests/fetch/test_fetch_ubuntu.py

```diff
@@ -730,6 +730,98 @@ def test_add_distro_source(self, log):
         # distro is a noop but test validate no exception is thrown
         fetch.add_source(source=source)
 
+    @patch('charmhelpers.fetch.ubuntu._add_cloud_pocket')
+    @patch('charmhelpers.fetch.ubuntu.get_distrib_codename')
+    def test_add_bare_openstack_is_distro(
+            self, mock_get_distrib_codename, mock_add_cloud_pocket):
+        mock_get_distrib_codename.return_value = 'focal'
+        fetch.add_source('ussuri')
+        mock_add_cloud_pocket.assert_not_called()
+
+    @patch('charmhelpers.fetch.ubuntu._add_cloud_pocket')
+    @patch('charmhelpers.fetch.ubuntu.get_distrib_codename')
+    def test_add_bare_openstack_is_cloud_pocket(
+            self, mock_get_distrib_codename, mock_add_cloud_pocket):
+        mock_get_distrib_codename.return_value = 'bionic'
+        fetch.add_source('ussuri')
+        mock_add_cloud_pocket.assert_called_once_with("bionic-ussuri")
+
+    @patch('charmhelpers.fetch.ubuntu._add_cloud_pocket')
+    @patch('charmhelpers.fetch.ubuntu.get_distrib_codename')
+    def test_add_bare_openstack_impossible_version(
+            self, mock_get_distrib_codename, mock_add_cloud_pocket):
+        mock_get_distrib_codename.return_value = 'xenial'
+        try:
```

Comment:
nitpick, you just reinvented this wheel:

```
with self.assertRaises(fetch.SourceConfigError):
    fetch.add_source('ussuri')
mock_add_cloud_pocket.assert_not_called()
```

Reply:
If you look at the rest of the file, it's the pattern that already exists.  Also, assertRaises() isn't actually available in this class for the bizarre reason that it uses testtools where it doesn't exist.  One of those oddities of the library.